text/0000-io.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: io
+- Start Date: 2016-07-09
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the addition of the io package to the standard library. This package will contain interfaces to address the common actions of reading and writing data. The ReadBuffer and WriteBuffer classes will also be moved from the net package into the new io package so that they are more easily discovered.
+
+# Motivation
+
+Most applications need to process some input and produce some output based on that input. Therefore, these operations should be raised above application specific packages so that they can be used in a generalized way.
+
+# Detailed design
+
+- The io package will contain the following interfaces:
+```pony
+interface Readable
+  """
+  Wraps the read method which returns up to len bytes from the input.
+  """
+  fun read(len: USize): Array[U8] iso^
```

Comment:
There's no type in the pony standard library so far that conforms to this interface.

`File.read` is close, but the receiver cap is `ref`, not `box`.